radio/src/targets/sky9x/usb/device/hid-joystick/HIDDJoystickDriver.c

```diff
@@ -0,0 +1,358 @@
+/// Sends the requested report to the host.
+/// \param type Report type.
+/// \param length Maximum number of bytes to send.
+//------------------------------------------------------------------------------
+static void HIDDJoystickDriver_GetReport(unsigned char type,
+                                         unsigned short length)
+{
+    TRACE_INFO("gReport ");
+
+    // Check report type
+    switch (type) {
+
+        case HIDReportRequest_INPUT:
+            TRACE_INFO("In ");
+
+            // Adjust size and send report
+            if (length > sizeof(HIDDJoystickInputReport)) {
+
+                length = sizeof(HIDDJoystickInputReport);
+            }
+            USBD_Write(0, // Endpoint #0
+                       &(hiddJoystickDriver.inputReport),
+                       length,
+                       0, // No callback
+                       0);
+            break;
+
+        default:
+            USBD_Stall(0);
+    }
+}
+
+//------------------------------------------------------------------------------
+/// Retrieves the new value of a report from the host and saves it.
+/// \param type Report type.
+/// \param length Report length.
+//------------------------------------------------------------------------------
+static void HIDDJoystickDriver_SetReport(unsigned char type,
+                                         unsigned short length)
+{
+    TRACE_INFO("sReport ");
+
+    // Check report type
+    switch (type) {
+    
+        case HIDReportRequest_INPUT:
+            // SET_REPORT requests on input reports are ignored
+            USBD_Stall(0);
+            break;
+
+        default:
+            USBD_Stall(0);
+    }
+}
+
+//------------------------------------------------------------------------------
+//         Optional RequestReceived() callback re-implementation
+//------------------------------------------------------------------------------
+#if !defined(NOAUTOCALLBACK)
+
+//------------------------------------------------------------------------------
+/// Callback function when new request receivce from host
+/// \param request Pointer to the USBGenericRequest instance
+//------------------------------------------------------------------------------
+void USBDCallbacks_RequestReceived(const USBGenericRequest *request)
+{
+    HIDDJoystickDriver_RequestHandler(request);
+}
+
+#endif
+
+//------------------------------------------------------------------------------
+//         ConfigurationChanged() callback re-implementation
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+/// Callback function when configureation changed
+/// \param cfgnum New configuration number
+//------------------------------------------------------------------------------
+void USBDDriverCallbacks_ConfigurationChanged(unsigned char cfgnum)
+{
+    if (cfgnum > 0) {
```

Comment:
Suspicious: empty if block and cfgnum is unsigned, so can never be negative so the test is meaningless.

Is this correct code? Should this have a comment on why it's ignored?  Does it need a trace?

Reply:
You're right, this is a meaningless construction. It's just part of the [standard example written by Atmel.](https://github.com/lfwanner/var/blob/e9802710380a480fa91a6660e754f3a087be5484/sam3/at91lib/usb/device/hid-mouse/HIDDMouseDriver.c) FYI, the same construction (the only difference is non-empty if block) [is already in the OpenTX](https://github.com/opentx/opentx/blob/2.2/radio/src/targets/sky9x/usb/device/massstorage/MSDDriver.c) :)

Fixed in 2efabc3